middleware.ts

```diff
@@ -0,0 +1,36 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { getTenantSlug } from './src/utils/multiTenancy/helpers';
+
+export const config = {
+  matcher: [
+    //     /*
+    //      * Match all paths except for:
+    //      * 1. /api routes
+    //      * 2. /_next (Next.js internals)
+    //      * 3. /_static (inside /public)
+    //      * 4. all root files inside /public (e.g. /favicon.ico)
+    //      */
```

Comment:
@Shreyaschorge Could you update this comment as well, so that it's easy to understand the matcher logic in the future.